nistats/utils.py

```diff
@@ -9,15 +9,12 @@
 
 py3 = sys.version_info[0] >= 3
 
-TINY = 1e-15
-
 
 def z_score(pvalue):
     """ Return the z-score corresponding to a given p-value.
     """
-    pvalue = np.minimum(np.maximum(pvalue, TINY), 1. - TINY)
-    z = norm.isf(pvalue)
-    return z
+    pvalue = np.minimum(np.maximum(pvalue, 1.e-300), 1. - 1.e-16)
```

Comment:
The purpose of TINY is to avoid numerical underlows. It turns out that the 1.e-15 used previously was not 'small' enough and created unpleasant numerical artifacts. I now use smaller values while still avoiding numerical instabilities.